tests/sqlparser_redshift.rs

```diff
@@ -196,3 +199,118 @@ fn test_create_view_with_no_schema_binding() {
     redshift_and_generic()
         .verified_stmt("CREATE VIEW myevent AS SELECT eventname FROM event WITH NO SCHEMA BINDING");
 }
+
+#[test]
+fn test_redshift_json_path() {
+    let sql = "SELECT cust.c_orders[0].o_orderkey FROM customer_orders_lineitem";
+    let select = redshift().verified_only_select(sql);
```

Comment:
Could we use `let dialect = all_dialects_where(|d| d.supports_partiql())` for the tests? thinking so any dialect that gets this feature can potentially reuse the same tests